store/orm/orm.go

```diff
@@ -454,3 +455,53 @@ func (orm *ORM) CreateSession(sr models.SessionRequest) (string, error) {
 func (orm *ORM) InitializeModel(klass interface{}) error {
 	return orm.InitBucket(klass)
 }
+
+// AllInBatches iterates over every single entry in the passed bucket without holding
+// the entire contents in memory, pulling down batches and streaming over each entry.
+// Be sure not to use the passed bucket parameter as it is used as a buffer.
+func (orm *ORM) AllInBatches(bucket interface{}, callback interface{}, optionalBatchSize ...int) error {
+	skip := 0
+	batchSize := 1000
+	if len(optionalBatchSize) > 0 {
+		batchSize = optionalBatchSize[0]
+	}
+
+	vcallback := reflect.ValueOf(callback)
+	tcallback := reflect.TypeOf(callback)
+	if tcallback.NumOut() != 1 || tcallback.Out(0).Kind() != reflect.Bool {
+		return errors.New("AllInBatches callback has incorrect function signature, must return bool")
```

Comment:
I tend to like to make these a public constant so they can be tested against easily. e.g:

```go
const (
  ErrorInvalidCallbackSignature = errors.New("...")
)
```